frontend/src/metabase/browse/components/CollectionBreadcrumbsWithTooltip.tsx

```diff
@@ -0,0 +1,218 @@
+  collection,
+  containerName,
+}: {
+  collection: Collection;
+  containerName: string;
+}) => {
+  const collections = (collection.effective_ancestors || []).concat(collection);
+  const pathString = collections
+    .map(coll => getCollectionName(coll))
+    .join(" / ");
+  const ellipsifyPath = collections.length > 2;
+  const shownCollections = ellipsifyPath
+    ? [collections[0], collections[collections.length - 1]]
+    : collections;
+  const justOneShown = shownCollections.length === 1;
+
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLAnchorElement>();
+
+  const initialEllipsisRef = useRef<HTMLDivElement | null>(null);
+  const [
+    isFirstCollectionDisplayedAsEllipsis,
+    setIsFirstCollectionDisplayedAsEllipsis,
+  ] = useState(false);
+
+  useLayoutEffect(() => {
+    const initialEllipsis = initialEllipsisRef.current;
+    if (!initialEllipsis) {
+      return;
+    }
+    const handleResize = () => {
+      // The initial ellipsis might be hidden via CSS,
+      // so we need to check whether it is displayed via getComputedStyle
+      const style = window.getComputedStyle(initialEllipsis);
+      setIsFirstCollectionDisplayedAsEllipsis(style.display !== "none");
+    };
+    resizeObserver.subscribe(initialEllipsis, handleResize);
+    return () => {
+      resizeObserver.unsubscribe(initialEllipsis, handleResize);
+    };
+  }, [initialEllipsisRef]);
+
+  const isTooltipEnabled =
+    areAnyTruncated || ellipsifyPath || isFirstCollectionDisplayedAsEllipsis;
+
+  const maxWidths = getBreadcrumbMaxWidths(shownCollections, 96, ellipsifyPath);
+
+  return (
+    <Tooltip
+      disabled={!isTooltipEnabled}
+      variant="multiline"
+      label={pathString}
+    >
+      <ResponsiveContainer
+        aria-label={pathString}
+        data-testid={`collection-breadcrumbs-for-collection: ${collection.name}`}
+        name={containerName}
+        w="auto"
+      >
+        <Flex align="center" w="100%" lh="1" style={{ flexFlow: "row nowrap" }}>
+          <FixedSizeIcon name="folder" style={{ marginInlineEnd: ".5rem" }} />
+          {shownCollections.map((collection, index) => {
+            const key = `collection${collection.id}`;
+            return (
+              <Group spacing={0} style={{ flexFlow: "row nowrap" }} key={key}>
+                {index > 0 && <PathSeparator />}
+                <CollectionBreadcrumbsWrapper
+                  containerName={containerName}
+                  style={{ alignItems: "center" }}
+                  w="auto"
+                  display="flex"
+                >
+                  {index === 0 && !justOneShown && (
+                    <Ellipsis
+                      ref={initialEllipsisRef}
+                      includeSep={false}
+                      className="initial-ellipsis"
+                    />
+                  )}
+                  {index > 0 && ellipsifyPath && <Ellipsis />}
+                  <Breadcrumb
```

Comment:
Because Breadcrumb is an `Anchor`, it does a full page refresh when clicking. Can we try using a `Link` component so that the navigation happens in app?

Reply:
See my comment above: https://github.com/metabase/metabase/pull/41960/files#r1587870441